manubot/process/util.py

```diff
@@ -168,12 +168,21 @@ def add_author_affiliations(variables):
     return variables
 
 
-def get_metadata_and_variables(args):
+def load_variables(args) -> dict:
     """
-    Process metadata.yaml and create variables available for jinja2 templating.
+    Read metadata.yaml and files specified by --template-variables-path to generate
+    manuscript variables available for jinja2 templating.
+
+    Returns a dictionary with the following keys:
+    - `pandoc`: a dictionary for passing options to Pandoc via the `yaml_metadata_block`.
+      All fields from a manuscript's `metadata.yaml` are copied to this dictionary,
+      expect for special Manubot-defined fields. Special fields include `author_info` and
+      `thumbnail`.
+    - `manubot`: a dictionary with manubot-generated variables.
+    - user-specified fields inserted according to the `--template-variables-path` option.
     """
     # Generated manuscript variables
-    variables = collections.OrderedDict()
+    variables = {"pandoc": {}, "manubot": {}}
```

Comment:
@dhimmel are these not OrderedDict because the regular dict is ordered in Python 3.6+?

Reply:
Yes and now Python has committed to keeping dicts ordered in future versions.

Before, our practice was to use OrderedDict to ensure dict order. But now since that's gauranteed going forward, we should use OrderedDict only when necessary to trigger a special behavior somewhere (like pandas sometimes treats dicts and ordereddicts separately). But I'm thinking we should default to using dict and only cast to OrderedDict when absolutely necessary. I imagine OrderedDict may become deprecated someday.